lib/puppet/pops/lookup/lookup_adapter.rb

```diff
@@ -68,7 +68,8 @@ def lookup(key, lookup_invocation, merge)
   end
 
   def lookup_global(key, lookup_invocation, merge_strategy)
-    terminus = Puppet[:data_binding_terminus]
+    # hiera_xxx will always use global_provider regardless of data_binding_terminus setting
+    terminus = lookup_invocation.hiera_xxx_call? ? :hiera : Puppet[:data_binding_terminus]
```

Comment:
I think this means that `lookup` and `hiera` will produce different results. That in turn means it is close to impossible to migrate from `hiera_xxx` to `lookup` calls or using modules using `lookup` if also having `data_binding_terminus` set to `none`. I think that the terminus setting should only affect APL.